acceptancetests/assess_persistent_storage.py

```diff
@@ -1,139 +1,428 @@
+    else:
+        storage_type = storage_output['storage'][storage_id]['kind']
+        # Be careful, the type of persistent_setting's value is boolean.
+        persistent_setting = storage_output['storage'][storage_id]['persistent']
+        if storage_type == 'block':
+            pool_storage = storage_output['volumes']['0']['storage']
+            pool_setting = storage_output['volumes']['0']['pool']
+            storage_status = storage_output['volumes']['0']['status']['current']
+        elif storage_type == 'filesystem':
+            pool_storage = storage_output['filesystems']['0/0']['storage']
+            pool_setting = storage_output['filesystems']['0/0']['pool']
+            storage_status = ''
+        else:
+            raise JujuAssertionError(
+                'Incorrect storage type. '\
+                'Found: {}\nExpected: block or filesystem'.format(storage_type))
+    return (
+        storage_list, storage_type,
+        persistent_setting, pool_storage, pool_setting, storage_status)
+
+
+def assert_app_status(client, charm_name, expected):
+    # Run juju status --format json
+    log.info('Checking current status of app {}...'.format(charm_name))
+    status_output = json.loads(
+        client.get_juju_output('status', '--format', 'json', include_e=False))
+    app_status = status_output['applications'][charm_name]['application-status']['current']
+
+    if app_status != expected:
+        raise JujuAssertionError(
+            'App status is incorrect. '\
+            'Found: {}\nExpected: {}\n.'.format(
+            app_status, expected))
+    else:
+        log.info('The current status of app {} is: {}; Expected: {}'.format(
+        charm_name, app_status, expected))
+
+
+def assert_storage_number(storage_list, expected):
+    log.info('Checking total number of storage unit(s)...')
+    found = len(storage_list)
+    if found != expected:
+        raise JujuAssertionError(
+            'Unexpected number of storage unit(s). '\
+            'Found: {}\nExpected: {}\n.'.format(
+            str(found), str(expected)))
+    else:
+        log.info(
+        'Found {} storage unit(s). Expected: {}.'.format(
+        str(found), str(expected)))
+
+
+def assert_single_blk_existence(storage_list, storage_id):
+    log.info(
+        'Checking existence of single block device storage...'.format(storage_id))
+    if storage_id not in storage_list:
+        raise JujuAssertionError(
+            '{} missing from storage list.'.format(storage_id))
+    else:
+        log.info('{} found in storage list'.format(storage_id))
+
+
+def assert_single_blk_removal(storage_list, storage_id):
+    log.info(
+        'Checking removal of single block device storage...'.format(storage_id))
+    if storage_id in storage_list:
+        raise JujuAssertionError(
+            '{} still exists in storage list.'.format(storage_id))
+    else:
+        log.info('{} has been removed from storage list'.format(storage_id))
```

Comment:
The messages I give to log.info() have 2 purposes, using human readable language to simply inform user a current status, and to use as a indicator in case of debug, e.g., if something go wrong, search keyword "been removed from storage list" in stdout (Ctrl+F), see if it exists. I personally prefer log a very simple line indicating success or failure, but not too much, and leave several "indicator" words there.